src/Protocol/Protocol.h

```diff
@@ -62,6 +63,10 @@ class cProtocol
 
 	/** Called when client sends some data */
 	virtual void DataReceived(const char * a_Data, size_t a_Size) = 0;
+	void ForgeDataReceived(const char * a_Data, size_t a_Size)
```

Comment:
I think it shouldn't need to be, the way I currently implemented it is the Forge protocol is implemented only within cForgeHandshake, not each cProtocol_X class (no need to override) -- it hasn't differed significantly between game protocol versions, any changes should be minor enough to handle within the cForge classes themselves, imho.